### ISSUE TITLE
[Bug] Can't override file name using attachFile

### ISSUE THREAD

**Issue by paul-grimmer** [abramenal/cypress-file-upload#180]
### Current behavior:

In v3, you could change the name of the file being loaded from the fixture like so:
```javascript
    const filePath = `video.mp4`;
    const fileName = `${String(Math.random())}.mp4`;
    cy.fixture(filePath, "base64").then(fileContent => {
      cy.get('input[accept=".wav,.mpg,.mpeg,.mov,.mxf,.mp4"]').upload(
        {
          fileContent,
          fileName,
          mimeType: "application/mxf",
          encoding: "base64",
        },
        { subjectType: "input" },
      );
    });
```
This was useful for us as our application validates that Filenames are Unique. Replacing the filename allowed us to get around this limitation. 

The New attachFile function does not accept a parameter for the filename. Meaning that we're stuck with the name of the file on the disk. 

### Desired behavior:

It would be good if attachFile accepted an optional fileName argument that would replace the name if provided.

### Versions

Cypress 4.3.0
Windows 10
Chrome 80

**Comment by bestKY** [comment]
In V3, the file rename is very useful for me, too.
I can upload the same file, but different filename, for mt e2e test.
I also hope to the V4 can has this feature as V3. Thanks.

**Comment by knotthere** [comment]
I too rely on being able to rename the file, so I'll wait for that before moving to v4.

**Comment by abramenal** [comment]
Hi, thanks for submitting the issue!

Right now I am not able to take a look into the issue properly.
In the meantime, I can suggest to check if the issue exist in v3 (previous major version):
https://github.com/abramenal/cypress-file-upload/tree/v3.5.3
Sorry about that. Not sure I can provide ETA on solving this issue in v4 since I am the only one contributor and maintainer. If you have ideas on fixing that, please feel free to open Pull Request and we can review it together.

Thanks for understanding.

**Comment by jdcl32** [comment]
I have submitted a PR fixing this and other issues: https://github.com/abramenal/cypress-file-upload/pull/184

**Comment by abramenal** [comment]
A change of @jdcl32 is released in `v4.0.5` so please take a look and let us know if something isn't working still.
